client/service/src/client/client.rs

```diff
@@ -1728,9 +1728,8 @@ where
 			.block_status(&BlockId::Hash(hash))
 			.map_err(|e| ConsensusError::ClientImport(e.to_string()))?
 		{
-			BlockStatus::InChainWithState | BlockStatus::Queued if !import_existing =>
+			BlockStatus::InChainWithState | BlockStatus::Queued =>
```

Comment:
This will break Cumulus.

Reply:
Cumulus really needs just to move the best block pointer without reimporting the whole thing. Surprisingly all tests still pass in cumulus with this change.

Reply:
> move the best block pointer without reimporting the whole thing

How?

Reply:
Client should expose `set_head` function that does that. It is really out of the scope of this PR so I'll work around it for  now and create a separate issue.

Reply:
Cumulus is fine with this change because it does not even call this function. Furthermore cumulus does not attempt to import the same block state twice. This is what this PR prevents from doing.